Backpack-SwiftUI/Dialog/Classes/BPKDialog.swift

```diff
@@ -0,0 +1,167 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public extension View {
+    /// Displays a success dialog with a title, text and a list of buttons.
+    func bpkSuccessDialog(
+        presented: Binding<Bool>,
+        title: String,
+        text: String,
+        confirmButton: BPKDialogButton,
+        secondaryButton: BPKDialogButton? = nil,
+        linkButton: BPKDialogButton? = nil,
```

Comment:
when talking to adam the link button should only be used when the secondary button is also available. on compose we therefore created another function for this which makes the secondary button required if link is provided

Reply:
done